hoot-core/src/main/cpp/hoot/core/geometry/GeometryUtils.cpp

```diff
@@ -228,20 +245,143 @@ Envelope GeometryUtils::envelopeFromConfigString(const QString& boundsStr)
   const QRegExp boundsRegEx("(-*\\d+\\.*\\d*,){3}-*\\d+\\.*\\d*");
   if (!boundsRegEx.exactMatch(boundsStr))
   {
-    throw HootException(errorMsg);
+    throw IllegalArgumentException(errorMsg);
   }
   const QStringList boundsParts = boundsStr.split(",");
   assert(boundsParts.size() == 4);
   if ((boundsParts.at(2).toDouble() <= boundsParts.at(0).toDouble()) ||
        boundsParts.at(3).toDouble() <= boundsParts.at(1).toDouble())
   {
-    throw HootException(errorMsg);
+    throw IllegalArgumentException(errorMsg);
   }
   return
     Envelope(boundsParts.at(0).toDouble(), boundsParts.at(2).toDouble(),
       boundsParts.at(1).toDouble(), boundsParts.at(3).toDouble());
 }
 
+std::shared_ptr<geos::geom::Polygon> GeometryUtils::envelopeToPolygon(
+  const geos::geom::Envelope& env)
+{
+  LOG_VART(env.isNull());
+  if (env.isNull())
+  {
+    return std::shared_ptr<geos::geom::Polygon>();
+  }
+
+  CoordinateSequence* coordSeq =
+    GeometryFactory::getDefaultInstance()->getCoordinateSequenceFactory()->create(5, 2);
+  coordSeq->setAt(geos::geom::Coordinate(env.getMinX(), env.getMinY()), 0);
+  coordSeq->setAt(geos::geom::Coordinate(env.getMinX(), env.getMaxY()), 1);
+  coordSeq->setAt(geos::geom::Coordinate(env.getMaxX(), env.getMaxY()), 2);
+  coordSeq->setAt(geos::geom::Coordinate(env.getMaxX(), env.getMinY()), 3);
+  coordSeq->setAt(geos::geom::Coordinate(env.getMinX(), env.getMinY()), 4);
+
+  // an empty set of holes
+  vector<Geometry*>* holes = new vector<Geometry*>();
+  // create the outer line
+  LinearRing* outer = GeometryFactory::getDefaultInstance()->createLinearRing(coordSeq);
+  std::shared_ptr<Polygon> poly(
+    GeometryFactory::getDefaultInstance()->createPolygon(outer, holes));
+  LOG_VAR(poly->isValid());
+  return poly;
+}
+
+std::shared_ptr<Polygon> GeometryUtils::polygonFromString(const QString& str)
+{
+  // format: x1,y1;x2,y2;x3,y3...
```

Comment:
This was a simple and readable way to do this for me. Can definitely change as needed to accommodate different geometries but works for now.